src/components/pages/index.js

```diff
@@ -22,6 +22,20 @@ function Index({ initialState }) {
   const currentHomeLanguage = homeLanguages[currentHomeLanguageIndex];
   const [isSearching, setIsSearching] = useState(false);
 
+  const handlePageShow = event => {
+    if (event.persisted) {
+      window.location.reload(true);
+    }
+  };
+
+  useEffect(() => {
+    window.addEventListener('pageshow', handlePageShow);
+
+    return () => {
+      window.removeEventListener('pageshow', handlePageShow);
+    };
+  }, []);
```

Comment:
You need to remove this empty dependency array, else it doesn't do the cleanup. I'm just surprised that the pageshow event doesn't trigger on the other pages. It's probably because it's a new page from the server side.

Reply:
With the empty array we want only do this on the initial render, we don't want to add the listener with every re-render.

Reply:
Fair, but then you can remove the de-registration, it doesn't get called at all.

Reply:
Like Andreas said here https://github.com/educandu/educandu/pull/33#discussion_r751395632 I'd keep this as good practice

Reply:
ok